contrib/inventory/azure_rm.py

```diff
@@ -306,7 +306,7 @@ def __init__(self, args):
         else:
             # try to look up "well-known" values via the name attribute on azure_cloud members
             all_clouds = [x[1] for x in inspect.getmembers(azure_cloud) if isinstance(x[1], azure_cloud.Cloud)]
-            matched_clouds = [x for x in all_clouds if x.name == raw_cloud_env]
+            matched_clouds = [x for x in all_clouds if x.name == raw_cloud_env.name]
             if len(matched_clouds) == 1:
                 self._cloud_environment = matched_clouds[0]
             elif len(matched_clouds) > 1:
```

Comment:
If the issue is corrected. There will be some cases triggering this branch, please also update the user cases of the `raw_cloud_env` variable.